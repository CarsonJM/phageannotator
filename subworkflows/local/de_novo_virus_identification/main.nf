//
// De novo identification of viral sequences in assemblies
//

include { GENOMAD_DOWNLOAD   } from '../../../modules/nf-core/genomad/download/main'
include { GENOMAD_ENDTOEND   } from '../../../modules/nf-core/genomad/endtoend/main'

workflow DE_NOVO_VIRUS_IDENTIFICATION {
    take:
    assemblies   // [ [ meta] , fasta    ], input assemblies (mandatory)

    main:
    ch_versions = Channel.empty()

    if ( params.genomad_db ){
        ch_genomad_db = file(params.genomad_db, checkIfExists: true)
    } else {
        ch_genomad_db = GENOMAD_DOWNLOAD ( ).genomad_db
        ch_versions = ch_versions.mix( GENOMAD_DOWNLOAD.out.versions )
    }

    GENOMAD_ENDTOEND ( assemblies, ch_genomad_db )
    ch_versions = ch_versions.mix( GENOMAD_ENDTOEND.out.versions )

    emit:
    identified_viruses = GENOMAD_ENDTOEND.out.virus_fasta   // [ [ meta ], fasta ]
    versions = ch_versions                                  // [ versions.yml ]

}
