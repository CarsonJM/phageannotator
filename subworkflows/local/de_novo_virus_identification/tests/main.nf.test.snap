{
    "genomad_end_to_end": {
        "content": [
            [
                "test1.contigs_virus.fna:md5,dc5f4c26f621ad61c0de55b086fc6c67",
                "test1.contigs_virus_genes.tsv:md5,430add59e3b16ec73b5705187b70bd4a",
                "test1.contigs_virus_proteins.faa:md5,abb89baa65716787a6fc37b263d0d3c9",
                "test1.contigs_virus_summary.tsv:md5,cfb67f95091666039f97e50c4091798b"
            ]
        ],
        "timestamp": "2023-08-14T20:16:02+0000"
    },
    "genomad_download": {
        "content": [
            "genomad_db:md5,b9ce17f8b01203c37a9147b5294dace6",
            "genomad_db.dbtype:md5,f2dd0dedb2c260419ece4a9e03b2e828",
            "genomad_db.index:md5,5693ab7d5fe2c7947a9a43a9b5705f55",
            "genomad_db.lookup:md5,973c8434e0cf70593af1f888c4424876",
            "genomad_db.source:md5,6805958c7c99a8e8ab38095a40dc0081",
            "genomad_db_h:md5,1325d5db1da081c479a455a3e7440ae4",
            "genomad_db_h.dbtype:md5,740bab4f9ec8808aedb68d6b1281aeb2",
            "genomad_db_h.index:md5,8c14ed8d3b520be4cd22340a250d26bf",
            "genomad_db_mapping:md5,2e10dbb66f99183ff42092777a8089ca",
            "genomad_db_taxonomy:md5,5abcff23ab2efa5f5dcfc37cc55ccbbc",
            "genomad_integrase_db:md5,23e19f271d2d65261588821a1e7d2b30",
            "genomad_integrase_db.dbtype:md5,f2dd0dedb2c260419ece4a9e03b2e828",
            "genomad_integrase_db.index:md5,7bae797a783d3f12c75845130ccb9b8a",
            "genomad_integrase_db.lookup:md5,cc0702fcafb9a0b27992fd54fc699d25",
            "genomad_integrase_db.source:md5,b6ff391f60a2d9d615c7eb41fbf01c19",
            "genomad_integrase_db_h:md5,118e59c1b7bbbdfa6776c17da3db430b",
            "genomad_integrase_db_h.dbtype:md5,740bab4f9ec8808aedb68d6b1281aeb2",
            "genomad_integrase_db_h.index:md5,2e8987568dae1a3f4c0207657fd84205",
            "genomad_marker_metadata.tsv:md5,400b9716a636f2fbfd18dd1c4a130486",
            "genomad_mini_db:md5,b9ce17f8b01203c37a9147b5294dace6",
            "genomad_mini_db.dbtype:md5,f2dd0dedb2c260419ece4a9e03b2e828",
            "genomad_mini_db.index:md5,e62e3de510ea7f5e73038f51ab02330f",
            "genomad_mini_db.lookup:md5,973c8434e0cf70593af1f888c4424876",
            "genomad_mini_db.source:md5,6805958c7c99a8e8ab38095a40dc0081",
            "genomad_mini_db_h:md5,1325d5db1da081c479a455a3e7440ae4",
            "genomad_mini_db_h.dbtype:md5,740bab4f9ec8808aedb68d6b1281aeb2",
            "genomad_mini_db_h.index:md5,8c14ed8d3b520be4cd22340a250d26bf",
            "genomad_mini_db_mapping:md5,2e10dbb66f99183ff42092777a8089ca",
            "genomad_mini_db_taxonomy:md5,5abcff23ab2efa5f5dcfc37cc55ccbbc",
            "mini_set_ids:md5,9cc65ecf08bba29cd75e436b18f49e4b",
            "names.dmp:md5,83f879df06a69bf16687b5de575ed5ba",
            "nodes.dmp:md5,ee4db67cdcf75a5bba6a17b83e57425d",
            "plasmid_hallmark_annotation.txt:md5,36bbd98d2f9e2f7abad5ef31aca1ce0c",
            "version.txt:md5,25198dbbb5f787214992c13a596f5751",
            "virus_hallmark_annotation.txt:md5,5211d849ade305ab989472a82c206700"
        ],
        "timestamp": "2023-08-14T20:16:02+0000"
    }
}